apps/remote_control/lib/lexical/remote_control/code_intelligence/rename/module.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Rename.Module do
+  alias Lexical.Ast
+  alias Lexical.Ast.Analysis
+  alias Lexical.Document
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Line
+  alias Lexical.Document.Position
+  alias Lexical.Document.Range
+  alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Store
+  require Logger
+
+  import Line
+
+  @spec rename(Analysis.t(), Position.t(), String.t()) ::
+          {:ok, %{Lexical.uri() => [Edit.t()]}} | {:error, term()}
+  def rename(%Analysis{} = analysis, %Position{} = position, new_name) do
```

Comment:
I'm considering solving this problem in the next PR. I'm not quite sure about the support for `RenameFile` in `documentChanges` across various editors. Additionally, can we also accomplish this through a code action approach?

Reply:
why would you not be sure about the support across editors? It's been there since LSP 3.13, and is part of a `WorkspaceEdit` message. If you're unsure about rename, you should be equally unsure about using `WorkspaceEdit`s. The spec also says

FYI, both LSP clients for emacs support workspace edits and document renames. I'm also certain vscode supports these as well. 

We can also investigate helix and zed rather easily, and you can check out nvim.

Reply:
Resovled in #651